### PR TITLE
Removed LoaderHeader.version === null for loading from just ops

### LINKED ISSUE #4049: Fix Loading From Ops
## Describe the bug
With detached container we always need to load from a snapshot to start. Currently if the container request passes `headers.version === null` we skip all snapshots, and load from ops only. This now causes a problem. We should update to load from the oldest snapshot when `headers.version === null`

https://github.com/microsoft/FluidFramework/blob/2b21b9dc5ab906a07924418e01d15bbf2d586237/packages/loader/container-loader/src/container.ts#L205
https://github.com/microsoft/FluidFramework/blob/2b21b9dc5ab906a07924418e01d15bbf2d586237/packages/loader/container-loader/src/container.ts#L1000

### PR DESCRIPTION
**Pull request by jatgarg**
Fixes: https://github.com/microsoft/FluidFramework/issues/4049
We always load from snapshot in load flow. So we can now remove version as null to just load from ops.

### CODE REVIEW

File: BREAKING.md [hunk 1]
```diff
@@ -10,6 +11,9 @@ the npm scopes.
 
 - `@fluidframework/last-edited-experimental` is renamed to `@fluid-experimental/last-edited`
 
+### LoaderHeader.version could not be null
+`LoaderHeader.version` in ILoader cann't be null as we always load from existing snapshot in `container.load()`;
```
Comment (vladsud) [inline BREAKING.md:15]: typo: cann't.

File: api-report/container-loader.api.md [hunk 2]
```diff
@@ -268,7 +268,7 @@ export interface IContainerLoadOptions {
     loadMode?: IContainerLoadMode;
     // (undocumented)
     resolvedUrl: IFluidResolvedUrl;
-    version?: string | null | undefined;
+    version?: string | undefined;
 }
```
Comment (vladsud) [inline api-report/container-loader.api.md:272]: undefined and ? are the same thing - can we use only one of them?

File: packages/loader/container-loader/src/loader.ts [hunk 3]
```diff
@@ -71,7 +71,7 @@ export class RelativeLoader implements ILoader {
                         canReconnect: request.headers?.[LoaderHeader.reconnect],
                         clientDetailsOverride: request.headers?.[LoaderHeader.clientDetails],
                         resolvedUrl: {...resolvedUrl},
-                        version: request.headers?.[LoaderHeader.version],
+                        version: request.headers?.[LoaderHeader.version] ?? undefined,
```
Comment (vladsud) [inline packages/loader/container-loader/src/loader.ts:74]: This change does not seem to be required

File: packages/loader/container-loader/src/loader.ts [hunk 4]
```diff
@@ -437,7 +432,7 @@ export class Loader implements IHostLoader {
                 canReconnect: request.headers?.[LoaderHeader.reconnect],
                 clientDetailsOverride: request.headers?.[LoaderHeader.clientDetails],
                 resolvedUrl: resolved,
-                version: request.headers?.[LoaderHeader.version],
+                version: request.headers?.[LoaderHeader.version] ?? undefined,
```
Comment (vladsud) [inline packages/loader/container-loader/src/loader.ts:435]: same here

**Review by vladsud — APPROVED** [review]

File: packages/loader/container-loader/src/loader.ts (hunk 4 above)
Comment (jatgarg) [reply packages/loader/container-loader/src/loader.ts:435]: request.headers?.[LoaderHeader.version] returns null if not present. So needs to return undefined.

File: packages/loader/container-loader/src/loader.ts (hunk 3 above)
Comment (jatgarg) [reply packages/loader/container-loader/src/loader.ts:74]: request.headers?.[LoaderHeader.version] returns null if not present. So needs to return undefined.

**Pull request merged**